src/api/v2/types.rs

```diff
@@ -237,29 +246,32 @@ impl Reply for SubscriptionId {
 pub enum RequestType {
 	Version,
 	Status,
+	Submit,
 }
 
 #[derive(Deserialize)]
 pub struct Request {
 	#[serde(rename = "type")]
 	pub request_type: RequestType,
 	pub request_id: String,
+	pub message: Option<Transaction>,
 }
```

Comment:
Why not to have request as externally tagged enum:
```rust
#[derive(Deserialize)]
#[serde(tag = "type")]
enum RequestInner {
    Version,
    Status,
    Submit {
        message: Transaction,
    },
}

#[derive(Deserialize)]
pub struct Request {
    pub request_id: String,
    #[serde(flatten)]
    pub inner: RequestInner,
}
```

Reply:
It would be great to ensure that every variant contains `message` field of different type if its needed, but Ill leave that for a moment when case appears.